src/decorators.ts

```diff
@@ -9,6 +9,17 @@
  * rights grant found at http://polymer.github.io/PATENTS.txt
  */
 
+export interface ElementConstructor extends Function {
```

Comment:
Can we also constrain this so that it has to extend HTMLElement? I'm thinking something like:

```typescript
export interface ElementConstructor<E extends HTMLElement> {
  new () => E;
  // .. everything else you've got here
}
```

That way we can get a compile time error if the user forgets to make their class extend HTMLElement:

```typescript
// this should fail!
@customElement('my-slider');
class MySlider { }

Reply:
Discussed offline. Tried some things out. It seems like decorator applications seem to upcast the constructor simply to Function, so the constraint can't be satisfied.